trace-obfuscation/src/sql.rs

```diff
@@ -0,0 +1,271 @@
+// Copyright 2023-Present Datadog, Inc. https://www.datadoghq.com/
+// SPDX-License-Identifier: Apache-2.0
+
+fn is_splitter(b: u8) -> bool {
+    matches!(
+        b,
+        b',' | b'(' | b')' | b'|' | b' ' | b'\t' | b'\n' | b'\r' |
+        0xB | // vertical tab 
+        0xC // form feed
+    )
+}
+
+fn is_numeric_litteral_prefix(bytes: &[u8], start: usize) -> bool {
+    matches!(
+        bytes[start],
+        b'0' | b'1' | b'2' | b'3' | b'4' | b'5' | b'6' | b'7' | b'8' | b'9' | b'-' | b'+' | b'.'
+    ) && !(start + 1 < bytes.len() && bytes[start] == b'-' && bytes[start + 1] == b'-')
+}
+
+fn is_hex_litteral_prefix(bytes: &[u8], start: usize, end: usize) -> bool {
+    (bytes[start] | b' ') == b'x' && start + 1 < end && bytes[start + 1] == b'\''
+}
+
+fn is_quoted(bytes: &[u8], start: usize, end: usize) -> bool {
+    bytes[start] == b'\'' && bytes[end - 1] == b'\''
+}
+
+/// Normalizes an sql string by replacing litterals with '?' chars.
```

Comment:
Confusing to have SQL normailization in the obfuscation project.

Reply:
The confusion spawns from the fact that I was following the java implementation which is named `SQLNormalizer`, when in practice, it is part of the obfuscation for trace stats in library spec

Reply:
Yeah, the line between `normalization` and `obfuscation` isn't that clear cut. I prefer to think about this as _normalizing_ the SQL statements when you remove _irrelevant_ (and maybe sensitive) data for the stats grouping, but let's not dwell on it.